File: csharp_body_deref/Gisketch.Aottg2UI.Runtime/Gisketch.Aottg2UI.Code/AottgToastDefinition.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Gisketch.Aottg2UI.Code.AottgToastDefinition
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Gisketch/Aottg2UI/Code/AottgToastDefinition.c
// Prior real C# source: none (type is newer than the last public source snapshot)
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using System.Runtime.InteropServices;
using Il2CppDummyDll;

namespace Gisketch.Aottg2UI.Code;

[Token(Token = "0x200006F")]
public sealed class AottgToastDefinition
{
	[Token(Token = "0x400023C")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x10")]
	public AottgToastVariant Variant;

	[Token(Token = "0x400023D")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x18")]
	public string Icon;

	[Token(Token = "0x400023E")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x20")]
	public string Title;

	[Token(Token = "0x400023F")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x28")]
	public string Content;

	[Token(Token = "0x4000240")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x30")]
	public string UpdateKey;

	[Token(Token = "0x60002EC")]
	[Address(RVA = "0x3A61700", Offset = "0x3A61700", VA = "0x3A61700")]
	public AottgToastDefinition(AottgToastVariant variant, string icon, string title, string content, [Optional] string updateKey)
	{
	}
}

File: cpp/Gisketch/Aottg2UI/Code/AottgToastDefinition.c
// Type: Gisketch.Aottg2UI.Code.AottgToastDefinition
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// C# structure: source/csharp/Gisketch.Aottg2UI.Runtime/Gisketch.Aottg2UI.Code/AottgToastDefinition.cs
// Prior source: NEW in this update
// --------------------------------

// Gisketch.Aottg2UI.Code.AottgToastDefinition$$.ctor
// il2cpp: void Gisketch_Aottg2UI_Code_AottgToastDefinition___ctor (Gisketch_Aottg2UI_Code_AottgToastDefinition_o* __this, int32_t variant, System_String_o* icon, System_String_o* title, System_String_o* content, System_String_o* updateKey, const MethodInfo* method);
// 0x3af6540

void Gisketch_Aottg2UI_Code_AottgToastDefinition___ctor
               (Gisketch_Aottg2UI_Code_AottgToastDefinition_o *__this,int32_t variant,
               System_String_o *icon,System_String_o *title,System_String_o *content,
               System_String_o *updateKey,MethodInfo *method)

{
  System_Object___ctor((Il2CppObject *)__this,(MethodInfo *)0x0);
  (__this->fields).Variant = variant;
  (__this->fields).Icon = icon;
  il2cpp_runtime_glue(&(__this->fields).Icon,icon);
  (__this->fields).Title = title;
  il2cpp_runtime_glue(&(__this->fields).Title,title);
  (__this->fields).Content = content;
  il2cpp_runtime_glue(&(__this->fields).Content,content);
  (__this->fields).UpdateKey = updateKey;
  il2cpp_runtime_glue(&(__this->fields).UpdateKey,updateKey);
  return;
}


